### PR TITLE
Update to current finagle (6.45.0) 

### PR DESCRIPTION
**Pull request by eugene-sy**
- [x] Updated finagle to 6.45.0
- [x] Dropped support for JVMv7
- [x] Dropped support for Scala 2.10.x

### CODE REVIEW

File: src/main/scala/com/twitter/finagle/WebSocket.scala [hunk 1]
```diff
@@ -2,43 +2,43 @@ package com.twitter.finagle
...
 object Websocket extends Server[Request, Response] {
-  case class Client(
-    stack: Stack[ServiceFactory[Request, Response]] = StackClient.newStack,
-    params: Stack.Params = StackClient.defaultParams + ProtocolLibrary("ws"))
+  case class Client(stack: Stack[ServiceFactory[Request, Response]] = StackClient.newStack,
+                    params: Stack.Params = StackClient.defaultParams + ProtocolLibrary("ws"))
   extends StdStackClient[Request, Response, Client] {
-    protected type In = Any
-    protected type Out = Any
+    override protected type In = Any
```
Comment (mkhq) [inline src/main/scala/com/twitter/finagle/WebSocket.scala:19]: Is the override necessary?

File: src/main/scala/com/twitter/finagle/WebSocket.scala [hunk 2]
```diff
@@ -2,43 +2,43 @@ package com.twitter.finagle
...
 
-    def withTls(cfg: Netty3TransporterTLSConfig): Client =
-      configured(Transport.TLSClientEngine(Some(cfg.newEngine))
-        ).configured(Transporter.TLSHostname(cfg.verifyHost))
+    def withTls(cfg: SslClientConfiguration): Client =
+      withTransport.tls.configured(Transport.ClientSsl(Some(cfg)))
   }
 
-  val client: Client = Client()
+  def client(): Client = Client()
```
Comment (mkhq) [inline src/main/scala/com/twitter/finagle/WebSocket.scala]: There is no need to create new instances of the Client case class for every new client, so it should be ok to keep the initial `val`

**Review by mkhq — COMMENTED** [review]
Looks good, just some small nitpicking

File: src/main/scala/com/twitter/finagle/WebSocket.scala (hunk 1 above)
Comment (eugene-sy) [reply src/main/scala/com/twitter/finagle/WebSocket.scala:19]: I believe it's better to use `override`. It adds checks at compile time so you know when method signature changed.

File: src/main/scala/com/twitter/finagle/WebSocket.scala (hunk 2 above)
Comment (eugene-sy) [reply src/main/scala/com/twitter/finagle/WebSocket.scala]: good catch

File: src/test/scala/com/twitter/finagle/websocket/EndToEndTest.scala [hunk 3]
```diff
@@ -46,7 +46,7 @@ private object EndToEndTest {
 
     val addr = server.boundAddress.asInstanceOf[InetSocketAddress]
 
-    val client = finagle.Websocket.client
+    val client = finagle.Websocket.client()
```
Comment (mkhq) [inline src/test/scala/com/twitter/finagle/websocket/EndToEndTest.scala]: since client is a val, the `()` should not be there

**Comment by mkhq** [comment]
Builds are failing because openjdk8 doesnt work with Travis: https://github.com/travis-ci/travis-ci/issues/6483

**Comment by eugene-sy** [comment]
That's interesting, it works for [builds from my fork](https://travis-ci.org/Axblade/finagle-websocket/builds/256809228)

**Comment by mkhq** [comment]
Ok, it looks like your fork is building with trusty already

**Comment by eugene-sy** [comment]
Shall we add `dist: trusty` to the travis-file?

**Comment by mkhq** [comment]
Yes, lets try that

**Pull request merged**